cassiopeia/preprocess/map_utils.py

```diff
@@ -0,0 +1,54 @@
+"""
+This file contains functions pertaining to mapping intBCs.
+Invoked through pipeline.py and supports the filter_molecule_table function.
+"""
+
+import pandas as pd
+
+from cassiopeia.mixins import logger
+from cassiopeia.preprocess import utilities
+
+
+@utilities.log_molecule_table
+def map_intbcs(molecule_table: pd.DataFrame) -> pd.DataFrame:
```

Comment:
You're right, I think we want to preferentially take the intBC with higher UMI count first, and then split by read count.